specification/machinelearningservices/AzureAI.Assets/model.tsp

```diff
@@ -0,0 +1,75 @@
+import "@typespec/rest";
+import "@azure-tools/typespec-autorest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-providerhub";
+import "@azure-tools/typespec-azure-core";
+import "./common.tsp";
+import "@typespec/openapi";
+
+using TypeSpec.OpenAPI;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+
+namespace Microsoft.MachineLearningServices;
+
+@doc("AssetVersion Definition")
+model AssetVersion {
+  @doc("Fully qualified resource Id: azureml://workspace/{workspaceName}/indexes/{name}/versions/{version} of the index.")
+  @visibility("read")
+  id: string;
+
+  @doc("Update stage to 'Archive' for soft delete. Default is Development, which means the asset is under development.")
+  stage: string = "Development";
+
+  @doc("Description information of the asset.")
+  description?: string;
+
+  @doc("Metadata containing createdBy and modifiedBy information.")
+  @visibility("read")
+  systemData?: SystemData;
+
+  @doc("Asset's tags.")
+  tags?: Record<string>;
+
+  @doc("Asset's properties.")
+  properties?: Record<string>;
+}
+
+@doc("Index resource Definition")
```

Comment:
Have you tried using @resource to define your REST resources and then use the standard Azure.Core resource operations?  This is usually how we would expect the TypeSpec of an Azure service, here's a getting started tutorial: https://azure.github.io/typespec-azure/docs/getstarted/azure-core/step04

Reply:
yes, my first commit followed the doc guidelines...I had to evolve it from there as I ran into issues